src/libtexture/texturesys.cpp

```diff
@@ -1514,25 +1522,29 @@ compute_miplevels(TextureSystemImpl::TextureFile& texturefile,
             aspect = OIIO::clamp(majorlength * r * 2.0f, 1.0f,
                                  float(options.anisotropic));
         }
+        return;
     }
     if (options.mipmode == TextureOpt::MipModeOneLevel) {
-        miplevel[0] = miplevel[1];
-        levelblend  = 0;
+        miplevel[0]    = miplevel[1];
+        levelweight[0] = 1.0f;
+        levelweight[1] = 0.0f;
+        return;
     }
-    if (options.mipmode == TextureOpt::MipModeStochasticTrilinear
-        || options.mipmode == TextureOpt::MipModeStochasticAniso) {
+    if (stochastic) {
         // If using stochastic sampling, the random deviate is a threshold
-        // versus the levelblend to determine which ONE of the two MIP
-        // levels to use.
-        if (options.rnd > levelblend) {
+        // versus the blend to determine which ONE of the two MIP levels to
+        // use. Then rescale options.rnd so we can use it again.
+        float blend = levelweight[1];
+        if (options.rnd >= blend) {
             miplevel[1] = miplevel[0];
+            options.rnd = (options.rnd - blend) / (1.0f - blend);
```

Comment:
Don't you wanna clamp here too?

Reply:
you mean clamp this computation to [0,1]? Sure, can do.

Reply:
updated